src/Voron/VoronExceptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Voron.Impl;
+
+namespace Voron
+{
+    public static class VoronExceptions
+    {
+        public static void ThrowIfReadOnly(LowLevelTransaction tx, string message = "A write transaction is required for this operation")
+        {
+            if (tx.Flags == TransactionFlags.Read)
```

Comment:
Doesn't it have any performance implications? I think we always wanted to do checks like that outside to avoid method calls

Reply:
OK